userguide/tutorials/notification_plugin.adoc

```diff
@@ -0,0 +1,11 @@
+= Developing a Notification Plugin
+
+== Overview
+
+Notification plugins implement the `OSGIKillbillEventDispatcher.OSGIKillbillEventHandler` interface. Each Kill Bill event from the external message bus is dispatched to the `public void handleKillbillEvent(final ExtBusEvent killbillEvent)` method. Such plugins can be a great alternative to http://docs.killbill.io/latest/push_notifications.html[Push Notifications] in performance demanding environments.
+
+== Retries
+
+By default, if your plugin triggers a runtime exception, the event is redispatched right away up to 3 times (or as configured by the `org.killbill.notificationq.external.max.failure.retry` global property). In some cases, you may want to retry at a later time (if for example a third-party provider is down). To do so, your plugin can throw a `QueueRetryException` with your own retry schedule.
+
```

Comment:
Should we detail a bit more how this exception + schedule works?

Reply:
Fixed by b437ad94d820f987780ce8b99b71fdcb8d10fd9e.